src/Contacts/src/CNLabeledValue.m

```diff
@@ -22,35 +22,69 @@
 @implementation CNLabeledValue
 
 - (instancetype)initWithLabel:(NSString *)label value:(id)value {
+    if (self = [super init]) {
+//        [self applyLabeledValueVariablesIdentifier: [[NSUUID alloc] init].UUIDString label:label value:value];
+        [self applyLabeledValueVariablesIdentifier: [[NSUUID alloc] init].UUIDString label:label value:value];
+    }
+    
+    return self;
+}
+
+- (instancetype)initWithIdentifier:(NSString *)identifier label:(NSString *)label value:(id)value {
+    if (self = [super init]) {
+        [self applyLabeledValueVariablesIdentifier:identifier label:label value:value];
+    }
+    
+    return self;
+}
+
+- (void)dealloc {
+    [_identifier release];
+    [_label release];
+    [_value release];
+    [super dealloc];
+}
+
+- (instancetype)applyLabeledValueVariablesIdentifier:(NSString *)identifier label:(NSString *)label value:(id)value {
+    if (value == nil) {
+        @throw [[NSException alloc] initWithName:@"CNPropertyInvalidTypeExpression" reason:@"Variable value cannot be null." userInfo:nil];
```

Comment:
Use [`[NSException raise:format:]`](https://developer.apple.com/documentation/foundation/nsexception/1569524-raise?language=objc)

Is "CNPropertyInvalidTypeExpression" the name that Apple's implementation raises?

Reply:
Yes, on MacOS you will get this exception message if you don't catch NSException:

```
*** Terminating app due to uncaught exception 'CNPropertyInvalidTypeException', reason: 'Labeled values require a nonnull value, but a null value was presented.'
```

Reply:
Does it look like Apple has a declaration like

```objc
const NSExceptionName CNPropertyInvalidTypeException = @"CNPropertyInvalidTypeException";
```

i.e. a symbol with the same name — or do they use a string literal directly? To check, you can run `nm` on the framework binary file.

Reply:
Do you know where Apple keeps their framework binary files?

Reply:
`/System/Library/Frameworks/Foo.framework/Foo`

Reply:
[Here is the entire output if you want to look at it](https://pastebin.com/ftDwsi2V).

The closest name I could find is `_CNPropertyInvalidTypeExceptionName`.

Reply:
Yeah, that sounds like it. The underscore is just Darwin name mangling, which means it's probably defined as

```objc
const NSExceptionName CNPropertyInvalidTypeExceptionName = @"CNPropertyInvalidTypeException";
```

Reply:
Where should I put it? It doesn't seem like you are able to publicly access this variable (XCode complains that it is a undeclared variable).

Reply:
That simply means it isn't declared in the header (only defined in the .m file)